lib/ecto/changeset.ex

```diff
@@ -1012,7 +1012,20 @@ defmodule Ecto.Changeset do
           "please use put_#{tag}/4 instead"
   end
 
-  defp put_change(data, changes, errors, valid?, key, value, _type) do
+  defp put_change(data, changes, errors, valid?, key, value, nil) do
```

Comment:
when a given `key` is not in the map of `types` (https://github.com/elixir-ecto/ecto/blob/v2.1.4/lib/ecto/changeset.ex#L1002). Without adding this clause, I got a bunch of failing tests